sycl/source/detail/memory_manager.cpp

```diff
@@ -322,11 +322,18 @@ void copyD2D(SYCLMemObjI *SYCLMemObj, RT::PiMem SrcMem, QueueImplPtr SrcQueue,
           Queue, SrcMem, DstMem, SrcOffset[0], DstOffset[0], SrcAccessRange[0],
           DepEvents.size(), &DepEvents[0], &OutEvent);
     } else {
+      bool isCUDA = Plugin.getBackend() == cl::sycl::backend::cuda;
+      size_t SrcSlicePitch, DstSlicePitch;
       size_t SrcRowPitch = (1 == DimSrc) ? 0 : SrcSize[0];
-      size_t SrcSlicePitch = (3 == DimSrc) ? SrcSize[0] * SrcSize[1] : 0;
-
       size_t DstRowPitch = (1 == DimDst) ? 0 : DstSize[0];
-      size_t DstSlicePitch = (3 == DimDst) ? DstSize[0] * DstSize[1] : 0;
+
+      if (!isCUDA) {
+        SrcSlicePitch = (3 == DimSrc) ? SrcSize[0] * SrcSize[1] : 0;
+        DstSlicePitch = (3 == DimDst) ? DstSize[0] * DstSize[1] : 0;
+      } else {
+        SrcSlicePitch = (DimSrc > 1) ? SrcSize[0] * SrcSize[1] : SrcSize[0];
```

Comment:
Since it is CUDA specific, can't we move it inside CUDA plugin?

Reply:
The problem is that it is not possible to determine the full size of the source and destination volumes from what is currently passed to PI API (CUDA requires this, but OpenCL seemingly does not).

I've updated this now so that the calculation is the same for all plugins - this doesn't break the OpenCL backend, and fixes the CUDA backend.